opentelemetry-sdk/src/opentelemetry/sdk/trace/export/__init__.py

```diff
@@ -370,15 +372,19 @@ class ConsoleSpanExporter(SpanExporter):
 
     def __init__(
         self,
+        service_name: Optional[str] = None,
         out: typing.IO = sys.stdout,
         formatter: typing.Callable[[Span], str] = lambda span: span.to_json()
         + os.linesep,
     ):
         self.out = out
         self.formatter = formatter
+        self.service_name = service_name
 
     def export(self, spans: typing.Sequence[Span]) -> SpanExportResult:
         for span in spans:
-            self.out.write(self.formatter(span))
+            span_json = json.loads(self.formatter(span))
+            span_json.update({"service_name": self.service_name})
+            self.out.write(json.dumps(span_json, indent=4))
```

Comment:
Is there any guarantee that the user will always provide a `formatter` that converts the span to JSON? Otherwise, a formatter like `lambda span: repr(span)` will break this logic.

I don't see an easy way to inject the service_name in the response of `span.to_json()`, unless the `Span` class is refactored, or `Span.attributes` is used (but that would mutate the Span object).

Reply:
@adamantike 
That is a good point. The only reason why we needed `service_name` is because the auto-instrumentation requires it to exist in the constructor of configured exporters [here](https://github.com/open-telemetry/opentelemetry-python/blob/master/opentelemetry-instrumentation/src/opentelemetry/instrumentation/auto_instrumentation/components.py#L81). We might want to revisit this restriction as to not have to enforce constructors to be a certain way. The usage of `service_name` is not mandatory, so for now you can just remove the assignment, but just keep it in the constructor.

Reply:
sure @lzchen will revert the changes.

Reply:
Thanks for the clarification, @lzchen!

Reply:
@lzchen, I have made the necessary changes. Could you please take a look?